rllib/evaluation/tests/test_env_runner_v2.py

```diff
@@ -92,6 +95,79 @@ def test_sample_batch_rollout_multi_agent_env(self):
         self.assertEqual(sample_batch.env_steps(), 200)
         self.assertEqual(sample_batch.agent_steps(), 400)
 
+    def test_guess_the_number_multi_agent(self):
+        """This test will test env runner in the game of GuessTheNumberGame.
+
+        The policies are chosen to be deterministic, so that we can test for an
+        expected reward. Agent 1 will always pick 1, and agent 2 will always guess that
+        the picked number is higher than 1. The game will end when the picked number is
+        1, and agent 1 will win. The reward will be 100 for winning, and 1 for each
+        step that the game is dragged on for. So the expected reward for agent 1 is 100
+        + 19 = 119.
+        """
+
+        register_env("env_under_test", lambda config: GuessTheNumberGame(config))
+
+        def mapping_fn(agent_id, *args, **kwargs):
+            return "pol1" if agent_id == 0 else "pol2"
+
+        class PickOne(RandomPolicy):
+            """This policy will always pick 1."""
+
+            def compute_actions(
+                self,
+                obs_batch,
+                state_batches=None,
+                prev_action_batch=None,
+                prev_reward_batch=None,
+                **kwargs
+            ):
+                return [np.array([2, 1])] * len(obs_batch), [], {}
+
+        class GuessHigherThanOne(RandomPolicy):
+            """This policy will guess that the picked number is higher than 1."""
+
+            def compute_actions(
+                self,
+                obs_batch,
+                state_batches=None,
+                prev_action_batch=None,
+                prev_reward_batch=None,
+                **kwargs
+            ):
+                return [np.array([1, 1])] * len(obs_batch), [], {}
+
+        config = (
+            PPOConfig()
+            .framework("torch")
+            .environment(disable_env_checking=True, env="env_under_test")
+            .rollouts(
+                num_envs_per_worker=1,
+                num_rollout_workers=0,
+                # Enable EnvRunnerV2.
+                enable_connectors=True,
+                rollout_fragment_length=100,
+            )
+            .multi_agent(
+                # this makes it independent of neural networks
+                policies={
+                    "pol1": PolicySpec(policy_class=PickOne),
+                    "pol2": PolicySpec(policy_class=GuessHigherThanOne),
+                },
+                policy_mapping_fn=mapping_fn,
+            )
+            .debugging(seed=42)
+        )
+
+        algo = PPO(config, env="env_under_test")
+
+        rollout_worker = algo.workers.local_worker()
+        sample_batch = rollout_worker.sample()
+        pol1_batch = sample_batch.policy_batches["pol1"]
+
+        # reward should be 100 (for winning) + 19 (for dragging the game for 19 steps)
+        check(pol1_batch["rewards"], 119 * np.ones_like(pol1_batch["rewards"]))
```

Comment:
can you check pol0's rewards here as well?
why do we get 119 reward for every single step of the batch? you didn't set gamma to 1.0?

Reply:
This is actually the expected behavior. For agents who receive a reward but it's not because of anything the did, you need to keep accumulating their reward until either the game ends (hence you generate a fake last obs with total reward) or they actually take an action that impacts the game.

Reply:
pol0's reward in this case it trivial: 19 steps of reward of 0 for every episode. So it's N_eps * 19 steps of zeros.

Reply:
oh, I see. that's interesting! can you check that agent 0 only gets a batch of 1 step, and that step has a reward of 119 then?

Reply:
done.